### PR TITLE
update apis for automatically propagate dependencies

### PR DESCRIPTION
**Pull request by mrlihanbo**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
update apis for automatically propagate dependencies according to Step 1 of #705 .

**Which issue(s) this PR fixes**:
Fixes Step 1 of #705 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
"NONE"

### CODE REVIEW

File: hack/.import-aliases [hunk 1]
```diff
@@ -42,17 +42,12 @@
     "k8s.io/apimachinery/pkg/api/errors": "apierrors",
     "k8s.io/apimachinery/pkg/apis/meta/v1": "metav1",
     "k8s.io/kubelet/apis/stats/v1alpha1": "kubeletstatsv1alpha1",
-    "k8s.io/kubernetes/pkg/controller/apis/config/v1alpha1": "controllerconfigv1alpha1",
-    "k8s.io/kubernetes/pkg/kubelet/apis/config/v1beta1": "kubeletconfigv1beta1",
```
Comment (RainbowMango) [inline hack/.import-aliases:46]: Please rebase your code. I remember we have removed this.

**Comment by RainbowMango** [comment]
```diff
diff --git a/pkg/apis/work/v1alpha2/binding_types.go b/pkg/apis/work/v1alpha2/binding_types.go
index 68d866af..df527cfa 100644
--- a/pkg/apis/work/v1alpha2/binding_types.go
+++ b/pkg/apis/work/v1alpha2/binding_types.go
@@ -35,8 +35,8 @@ type ResourceBindingSpec struct {
        Resource ObjectReference `json:"resource"`

        // PropagateDeps tells if relevant resources should be propagated automatically.
-       // e.g. a ConfigMap referred by a Deployment.
-       // default false.
+       // It is inherited from PropagationPolicy or ClusterPropagationPolicy.
+       // Defaults to false.
        // +optional
        PropagateDeps bool `json:"propagateDeps,omitempty"`

@@ -52,14 +52,11 @@ type ResourceBindingSpec struct {
        // +optional
        Clusters []TargetCluster `json:"clusters,omitempty"`

-       // TrackClusters stores the schedule results of resources which refer the current resource.
+       // RequiredBy represents the list of Bindings that depend on the referencing resource.
        // +optional
-       TrackClusters map[string]TargetClusters `json:"trackClusters,omitempty"`
+       RequiredBy []BindingSnapshot `json:"requiredBy,omitempty"`
 }

-// TargetClusters represents the list of TargetCluster
-type TargetClusters []TargetCluster
-
 // ObjectReference contains enough information to locate the referenced object inside current cluster.
 type ObjectReference struct {
        // APIVersion represents the API version of the referent.
@@ -125,6 +122,23 @@ type TargetCluster struct {
        Replicas int32 `json:"replicas,omitempty"`
 }

+// BindingSnapshot is a snapshot of a ResourceBinding or ClusterResourceBinding.
+type BindingSnapshot struct {
+       // Namespace represents the namespace of the Binding.
+       // It is required for ResourceBinding.
+       // If Namespace is not specified, means the referencing is ClusterResourceBinding.
+       // +optional
+       Namespace string `json:"namespace,omitempty"`
+
+       // Name represents the name of the Binding.
+       // +required
+       Name string `json:"name"`
+
+       // Clusters represents the scheduled result.
+       // +optional
+       Clusters []TargetCluster `json:"clusters,omitempty"`
+}
+
 // ResourceBindingStatus represents the overall status of the strategy as well as the referenced resources.
 type ResourceBindingStatus struct {
        // Conditions contain the different condition statuses.
```

**Review by GitHubxsy — APPROVED** [review]
/lgtm

**Pull request merged**